runtime/onert/backend/train/ops/PoolLayer.cc

```diff
@@ -35,36 +145,52 @@ void PoolLayer::configure(const IPortableTensor *input, const uint32_t paddingLe
                           const uint32_t paddingBottom, const uint32_t strideWidth,
                           const uint32_t strideHeight, const uint32_t kernelWidth,
                           const uint32_t kernelHeight, const ir::Activation activation,
-                          IPortableTensor *output, const PoolType op_type)
+                          IPortableTensor *output, const PoolType op_type,
+                          IPortableTensor *deriv_input, const IPortableTensor *deriv_output)
 {
+  _input = input;
+  _output = output;
+
+  _deriv_output = deriv_output;
+  _deriv_input = deriv_input;
+
+  // ready to infer
+  cpu::ops::PoolLayer::configure(input, paddingLeft, paddingRight, paddingTop, paddingBottom,
+                                 strideWidth, strideHeight, kernelWidth, kernelHeight, activation,
+                                 output, convert_to_infer(op_type));
+
+  // ready to train
+  if (output->data_type() != OperandType::FLOAT32)
+  {
+    throw std::runtime_error("PoolLayer : Unsupported data type for training");
+  }
```

Comment:
Both Inference and training kernel be ready in compile time. 
And In Runtime, depends on `bool training` value, It decides which kernel to use.